frontend/src/components/ChakraDataTable.tsx

```diff
@@ -89,7 +94,83 @@ interface ChakraDataTableProps<T> {
 }
 
 const TableWrapper = styled.div`
-  margin: 1rem 0;
+  margin-bottom: 3rem;
+`;
+
+const InputWrapper = styled.div`
+  margin-bottom: 2rem;
+`;
+
+const ColumnHeader = styled(Table.ColumnHeader)`
+  &.sortable {
+    cursor: pointer;
+  }
+
+  &.not-sortable {
+    cursor: default;
+  }
+`;
+
+const TableRow = styled(Table.Row)`
+  &.clickable {
+    cursor: pointer;
+  }
+
+  &.not-clickable {
+    cursor: default;
+  }
+`;
+
+const PaginationButton = styled(Button)`
+  background-color: #000;
+  color: #fff;
+`;
+
+const PaginationIconButton = styled(IconButton)`
+  background-color: transparent !important;
+
+  &:hover:not(:disabled) {
+    background-color: transparent !important;
+  }
+
+  &:active:not(:disabled) {
+    background-color: transparent !important;
+  }
+`;
+
+const HeaderContent = styled.div`
+  display: flex;
+  gap: 0.5rem;
+  align-items: center;
+`;
+
+const PaginationContentWrapper = styled.div`
+  display: flex;
+  gap: 0.5rem;
+`;
+
+const PaginationWrapper = styled.div`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  gap: 0.5rem;
+  margin-top: 1rem;
+`;
```

Comment:
With so many pagination specific components and styles I would suggest moving it to its own pagination component